xwiki-rendering-syntaxes/xwiki-rendering-syntax-xwiki20/src/main/java/org/xwiki/rendering/internal/renderer/xwiki20/reference/InterWikiReferenceTypeSerializer.java

```diff
@@ -45,27 +45,27 @@
      * Escapes to add when rendering a link reference part.
      */
     private static final String[] ESCAPE_REPLACEMENTS_REFERENCE = new String[] {
-        "" + XWiki20LinkReferenceParser.ESCAPE_CHAR + XWiki20LinkReferenceParser.ESCAPE_CHAR };
+        String.valueOf(XWiki20LinkReferenceParser.ESCAPE_CHAR + XWiki20LinkReferenceParser.ESCAPE_CHAR) };
```

Comment:
Should be `String.valueOf(XWiki20LinkReferenceParser.ESCAPE_CHAR) + XWiki20LinkReferenceParser.ESCAPE_CHAR` I think

Reply:
you're right - careless mistake on my part. will fix, thanks

Reply:
I don't fully agree with Sonarqube here and I find the original approach with the quote simpler to read and less verbose. In addition the compiler will very likely transform it into a performant solution. But I'm ok with the last solution proposed by Thomas, even though I still find the quote solution more readable. I'll let you guys decide what we want to do here but it's likely that I will do it again the way I wrote it initially.

Reply:
Some info here: https://stackoverflow.com/questions/15046668/preferred-way-of-declaring-and-converting-primitives-to-strings